index.js

```diff
@@ -1,16 +1,55 @@
+/// <reference path="index.d.ts" />
 'use strict';
 const path = require('path');
 const {constants: fsConstants} = require('fs');
 const pEvent = require('p-event');
 const CpFileError = require('./cp-file-error');
 const fs = require('./fs');
 const ProgressEmitter = require('./progress-emitter');
+const {version} = process;
 
+const defaultOptions = {
+	overwrite: true,
+	clone: true
+};
+
+/**
+ * @param {import("fs").PathLike} source
+ * @param {import("fs").PathLike} destination
+ */
+const updateStats = async (source, destination) => {
+	const stats = await fs.lstat(source);
+
+	return Promise.all([
+		fs.utimes(destination, stats.atime, stats.mtime),
+		fs.chmod(destination, stats.mode)
+	]);
+};
+
+/**
+ * @param {string} source
+ * @param {string} destination
+ * @param {import('.').Options} options
+ * @param {ProgressEmitter} progressEmitter
+ */
```

Comment:
I think all the JSDoc comments should have been added in a separate PR. Try to keep that in mind in the future. It makes it much easier to review without all the unrelated noise.